docs/book/src/SUMMARY.md

```diff
@@ -58,20 +58,15 @@
 - [Migrations](./migrations.md)
 
   - [Kubebuilder v1 vs v2](./migration/v1vsv2.md)
-
       - [Migration Guide](./migration/legacy/migration_guide_v1tov2.md)
```

Comment:
there's now a bunch of pages called some form of "migration guide".

I'd suggest we call this first one `Upgrading everything`, call the "project version only" guide `Staying on go/v2`.  It's not entirely clear to me what the difference is between the first "migration guide" and the v2 to v3 plugin migration guide.  I'd suggest we unify them, or maybe make it clear that one is the "copy & paste" method & the other is a very fiddly by-hand method.  That being said, it's not clear when you'd want to use the fiddly-by-hand vs the "regenerate the scaffolding", so maybe we just unify the two.